packages/opencensus-web-instrumentation-zone/test/test-interaction-tracker.ts

```diff
@@ -0,0 +1,408 @@
+    const onclick = () => {
+      // No change in task count as the click is not tracked.
+      expect(incrementTaskCountSpy.calls.count()).toBe(0);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+
+      setTimeout(() => {
+        // No change in task count as the click is not tracked.
+        expect(incrementTaskCountSpy.calls.count()).toBe(0);
+        expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      });
+    };
+    const button = createButton('test interaction', true);
+    fakeInteraction(onclick, button);
+
+    const onclick2 = () => {
+      // Click is detected as this button does is not 'disabled'.
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      setTimeout(() => {
+        expect(incrementTaskCountSpy.calls.count()).toBe(2);
+        expect(decrementTaskCountSpy.calls.count()).toBe(1);
+      });
+    };
+
+    button.removeAttribute('disabled');
+    fakeInteraction(onclick2, button);
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      // There should be only 2 counted tasks related to the second interaction.
+      expect(incrementTaskCountSpy.calls.count()).toBe(2);
+      expect(decrementTaskCountSpy.calls.count()).toBe(2);
+      expect(rootSpan.name).toBe('test interaction');
+      expect(rootSpan.ended).toBeTruthy();
+      expect(rootSpan.duration).toBeLessThanOrEqual(8);
+      done();
+    });
+  });
+
+  it('should handle interactions on elements without data-ocweb-id attribute', done => {
+    const onclick = () => {
+      setTimeout(noop, 1);
+    };
+    const button = createButton('');
+    fakeInteraction(onclick, button);
+
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      expect(rootSpan.name).toBe("<BUTTON> id:'test_element' click");
+      expect(rootSpan.ended).toBeTruthy();
+      expect(rootSpan.duration).toBeLessThanOrEqual(5);
+      done();
+    });
+  });
+
+  it('should ignore periodic tasks', done => {
+    const onclick = () => {
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      const interaval = setInterval(() => {
+        // The counted tasks does not change as this is ignored.
+        expect(incrementTaskCountSpy.calls.count()).toBe(1);
+        expect(decrementTaskCountSpy.calls.count()).toBe(1);
+        clearInterval(interaval);
+      }, 1);
+    };
+    fakeInteraction(onclick);
+
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(1);
+      expect(rootSpan.name).toBe('test interaction');
+      expect(rootSpan.ended).toBeTruthy();
+      expect(rootSpan.duration).toBeLessThanOrEqual(10);
+      done();
+    });
+  });
+
+  it('should not create a new interaction when second click occurs before 50 ms.', done => {
```

Comment:
Should this say something like "when second click *handler* occurs before 50 ms."? (May need to reword a bit to keep < 80 chars)